src/core/ipc_handler/unix_socket.rs

```diff
@@ -40,16 +40,20 @@ impl Connect for Handler {
 
 impl Handler {
     /// Create new client using given socket path and timeout duration
-    pub fn new(path: PathBuf, timeout: Option<Duration>) -> Self {
-        Handler { path, timeout }
+    pub fn new(path: PathBuf, timeout: Option<Duration>) -> Result<Self> {
+        if path.exists() {
```

Comment:
I think you also need to check if `path` points to a socket as we do [in the service](https://github.com/parallaxsecond/parsec/blob/master/src/front/domain_socket.rs#L42), it might be a folder or a regular file:
```rust
if path.exists() && fs::metadata(&path)?.file_type().is_socket() {
    Ok(...)
} else {
    Err(...)
}
```